cli/src/modules/mod.rs

```diff
@@ -0,0 +1,347 @@
+
+        if run_build {
+            // Run the build
+            let build_result = self.perform_build(prod).await;
+            // Update the build state and notify all the senders
+            {
+                let mut states = build_states.states_map.lock().unwrap();
+
+                let res_clone = match &build_result {
+                    Ok(spawn_res) => Ok(spawn_res.clone()),
+                    Err(err) => Err(anyhow::anyhow!("{}", err)),
+                };
+
+                let Some(BuildState::Running(senders)) =
+                    states.insert(target, BuildState::Finished(build_result))
+                else {
+                    unreachable!("State after calling build must be renning");
+                };
+
+                for sender in senders {
+                    let res_clone = match &res_clone {
+                        Ok(spawn_res) => Ok(spawn_res.clone()),
+                        Err(err) => Err(anyhow::anyhow!("{}", err)),
+                    };
+                    if sender.send(res_clone).is_err() {
+                        bail!("Fail to communicate with builder");
+                    }
+                }
+            }
+        }
+
+        rx_result
+            .await
+            .context("Fail to communicate with builder")?
+    }
+
+    /// Performs build process without checking the current builds states
+    async fn perform_build(&self, prod: bool) -> Result<Vec<SpawnResult>, Error> {
+        let mut results = Vec::new();
+        let install_result = self.install(false).await?;
+        results.push(install_result);
+        let deps: Vec<Box<dyn Manager + Sync + Send>> =
+            self.deps().iter().map(|target| target.get()).collect();
+        for module in deps {
+            let status = module.build(prod).await?;
+            results.extend(status);
+            if results.iter().any(|res| !res.status.success()) {
+                return Ok(results);
+            }
+        }
+        let path = get_root().join(self.cwd());
+        let cmd = self
+            .build_cmd(prod)
+            .unwrap_or_else(|| self.kind().build_cmd(prod));
+        let caption = format!("Build {}", self.owner());
+        match spawn(cmd, Some(path), caption, iter::empty(), None).await {
+            Ok(status) => {
+                if !status.status.success() {
+                    results.push(status);
+                    Ok(results)
+                } else {
+                    results.push(status);
+                    let res = self.after(prod).await?;
+                    if let Some(result) = res {
+                        results.push(result);
+                    }
+                    if matches!(self.kind(), Kind::Ts) && prod {
+                        let clean_res = self.clean().await?;
+                        results.push(clean_res);
+                        let install_res = self.install(prod).await?;
+                        results.push(install_res);
+                    }
```

Comment:
@DmitryAstafyev 
I was comparing the current ruby implementation and I couldn't find the equivalent of this code there, and I want to double check if we need it here.

Reply:
Hello @DmitryAstafyev Thanks for the clarification. 
I've noticed that this code will be called after calling `self.after()` which would actually do the manual copying. Do we need to move it to above the `self.after()` block?

Reply:
I moved that part to before copying the data in order to get the benefits from it